StateT.md

```diff
@@ -0,0 +1,300 @@
+
+---
+
+# StateT :: Example
+
+```kotlin
+fun goDownS(floors: Int = 1) = StateT<EitherPartialOf<ElevatorError>, Elevator, Int>(Either.monad()) { 
+	elevator: Elevator ->
+	    val tentativeFloor = elevator.currentFloor - floors
+
+	    if (tentativeFloor < 0) {
+	        ElevatorError.ElevatorDownError.left()
+	    } else {
+	        (Elevator(elevator.floors, tentativeFloor) toT tentativeFloor).right()
+	    }
+}
+```
+
+^ The same is applied here to our go down operation. Notice how we haven't had to change anything in our implementation of the operation to work with a Monad Transformer like StateT.
+
+---
+
+# StateT :: Monad instance
+
+We can use `flatMap` to combine the previous operations:
+
+```kotlin
+fun elevatorOpsFlatMap(): StateT<EitherPartialOf<ElevatorError>, Elevator, Int> {
+    val eitherMonad = Either.monad<ElevatorError>()
+    return liftUpS(2).flatMap(eitherMonad) { _ ->
+        goDownS(1).flatMap(eitherMonad) { _ ->
+            goDownS(1)
+        }
+    }
+}
+
+val resultFM = elevatorOpsFlatMap().runM(Either.monad(), Elevator(3, 0))
+val resultErrorFM  = elevatorOpsFlatMap().runM(Either.monad(), Elevator(1, 0))
+
+// resultFM = Right(b=Tuple2(a=Elevator(floors=3, currentFloor=0), b=0))
+// resultErrorFM = Left(a=ElevatorError$ElevatorLiftError@214c265e)
+```
+
+^ Monad Transformers are monads too, so we can use flatMap to combine the previous operations.
+Notice how we don't worry about being under the Either context or an State context. Each call will update the original state of our Elevator. See how  both in the success and error cases we get the capabilities of Either to handle errors in our operations, together with the ability of keeping state in a functional way like with State.
+
+---
+
+# StateT :: Monad binding
+
+```kotlin
+fun elevatorOpsS() = 
+	StateT.monad<EitherPartialOf<ElevatorError>, Elevator>(Either.monad()).binding {
+	    liftUpS(2).bind()
+	    goDownS(1).bind()
+	    val result = goDownS(1).bind()
+	    result
+}
+
+val result = elevatorOpsS().runM(Either.monad(), Elevator(3, 0))
+val resultError = elevatorOpsS().runM(Either.monad(), Elevator(1, 0))
+
+// result = Right(b=Tuple2(a=Elevator(floors=3, currentFloor=0), b=0))
+// resultError = Left(a=ElevatorError$ElevatorLiftError@214c265e)
+```
+
+Each call to bind() is a coroutine suspended function which will bind to it's value after each state has been updated to their new values.
+
+^ Being a Monad instance it means we can combine operations with our `StateT` instances in a more imperative-looking way with monad bindings. Each call to bind() is a coroutine suspended function which will bind to its value after each state has been updated to their new values.
+
+---
+
+# StateT :: Map transformations
+
+We can also use `map` to define transformations on our values before updating our state:
+
+```kotlin
```

Comment:
Missing an example with Applicative `map` with multiple operations which is what would happen uf a user presses 4 buttons at the same time when getting into the elevator.
```
val context = StateT.monad<EitherPartialOf<ElevatorError>, Elevator>(Either.monad())
with (context) {
  map(op1, op2, op3, op4, { ... })
}
```